torch/_refs/__init__.py

```diff
@@ -4299,17 +4300,8 @@ def arange(
     type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT,
 )
 def lerp(start: Tensor, end: Tensor, weight: Union[Tensor, NumberType]):
-    check(
```

Comment:
why no type checks here?

Reply:
They weren't necessary because all the operations below do correct type cast. I used to use this operation in linspace, and I relaxed the preconditions because I needed to do so to use it in `lerp`, but even now, I still think it'd be good to have the more general definition.